### ISSUE TITLE
Updating a binary causes a CastException

### ISSUE THREAD

**Issue by TomasEkeli** [ThatRendle/Simple.Data#49]
Updating a binary in MS-SQL seems to fail with a cast exception. I've tried this with Content set both to type image and varbinary(max).

```
using System.IO;
using NUnit.Framework;

namespace Simple.Data.ByteBug {
    [TestFixture]
    public class ByteBug {
        private const string connectionString = "server=.;database=simpleDataTest;uid=testuser;pwd=testpassword;";

        /*
         * --SQL definition of table Logo
         * CREATE TABLE dbo.Logo (SiteId int NOT NULL, Name nvarchar(50) NOT NULL, MimeType nvarchar(50) NOT NULL, Content] image NOT NULL)  ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
         * GO
         * ALTER TABLE dbo.Logo ADD CONSTRAINT PK_Logo PRIMARY KEY CLUSTERED  (SiteId) WITH(STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
         * GO 
         * ALTER TABLE dbo.Logo SET (LOCK_ESCALATION = TABLE)
         * GO
         */

        [Test(Description = "Updating an existing logo is possible")]
        public void Save_AnExistingLogo_Possible() {
            const string fileName1 = "document-new.png"; //doesn't matter which - these were from the Tango project
            const string fileName2 = "document-open.png";

            var bytes = getBytes(fileName1);

            var itemToSave = new Logo {
                MimeType = "image/png",
                Name = "testlogo.png",
                SiteId = 1,
                Content = bytes
            };

            var database = Database.OpenConnection(connectionString);

            database.Logo.DeleteBySiteId(1); //clean-up, if necessary

            database.Logo.Insert(itemToSave);

            var persisted = database.Logo.FindBySiteId(1);

            persisted.Content = getBytes(fileName2);

            database.Logo.Update(persisted);

            /*
             * failed: System.InvalidCastException : Failed to convert parameter value from a Byte to a Byte[].
             * ----> System.InvalidCastException : Invalid cast from 'System.Byte' to 'System.Byte[]'.
             * at System.Data.SqlClient.SqlParameter.CoerceValue(Object value, MetaType destinationType)
             * at System.Data.SqlClient.SqlParameter.GetCoercedValue()
             * at System.Data.SqlClient.SqlParameter.Validate(Int32 index, Boolean isCommandProc)
             * at System.Data.SqlClient.SqlCommand.BuildParamList(TdsParser parser, SqlParameterCollection parameters)
             * at System.Data.SqlClient.SqlCommand.BuildExecuteSql(CommandBehavior behavior, String commandText, SqlParameterCollection parameters, _SqlRPC& rpc)
             * at System.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean async)
             * at System.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method, DbAsyncResult result)
             * at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(DbAsyncResult result, String methodName, Boolean sendToPipe)
             * at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
             * at Simple.Data.Ado.AdoAdapter.TryExecute(IDbCommand command)
             * at Simple.Data.Ado.AdoAdapter.Execute(ICommandBuilder commandBuilder)
             * at Simple.Data.Ado.AdoAdapter.Update(String tableName, IDictionary`2 data, SimpleExpression criteria)
             * at Simple.Data.Database.Update(String tableName, IDictionary`2 data, SimpleExpression criteria)
             * at Simple.Data.Commands.UpdateCommand.UpdateByKeyFields(String tableName, DataStrategy dataStrategy, Object entity, IEnumerable`1 keyFieldNames)
             * at Simple.Data.Commands.UpdateCommand.Execute(DataStrategy dataStrategy, DynamicTable table, InvokeMemberBinder binder, Object[] args)
             * at Simple.Data.DynamicTable.TryInvokeMember(InvokeMemberBinder binder, Object[] args, Object& result)
             * at CallSite.Target(Closure , CallSite , Object , Object )
             * at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
             * ByteBug.cs(43,0): at Simple.Data.ByteBug.ByteBug.Save_AnExistingLogo_Possible()
             * --InvalidCastException
             * at System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
             * at System.Byte.System.IConvertible.ToType(Type type, IFormatProvider provider)
             * at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
             * at System.Data.SqlClient.SqlParameter.CoerceValue(Object value, MetaType destinationType)
             */
        }

        private byte[] getBytes(string fileName) {
            var file = new FileInfo(fileName);
            var fileLength = file.Length;
            var imageStream = new FileStream(file.Name, FileMode.Open, FileAccess.Read);
            return new BinaryReader(imageStream).ReadBytes((int)fileLength);
        }
    }

    public class Logo {
        public string MimeType { get; set; }
        public string Name { get; set; }
        public int SiteId { get; set; }
        public byte[] Content { get; set; }
    }
}
```

**Comment by ThatRendle** [comment]
Thank you for that detail! I will try to get this fixed for the next release, which should be going out this weekend.

**Comment by TomasEkeli** [comment]
That would be very impressive indeed! Thanks in advance!

**Comment by ThatRendle** [comment]
Sorry, didn't get time yesterday, but I've fixed this now and it's in the 0.6.3 release just pushed to NuGet.
